lib/deploy_complexity/deploy.rb

```diff
@@ -155,5 +144,49 @@ def pull_requests(merges)
         end
       end.compact
     end
+
+    def get_merges(commits)
+      merges = commits.grep(/Merges|\#\d+/)
+      merges = merges.select { |m| makes_changes_to(m) }
+      merges
+    end
+
+    def get_changed_files(range)
+      files = `git diff --name-only #{range}`
+      filtered = files.split(/\n/).select { |f| pattern =~ f }.join('\n')
+      DeployComplexity::ChangedFiles.new(filtered)
+    end
+
+    def makes_changes_to(merge)
+      commithash = merge.split(' ')[0]
+      commits = `git log -m -1 --name-only --first-parent --pretty="format:" #{commithash}`.split(/\n/)
+      commits.any? { |commit| pattern =~ commit }
+    end
+
+    def pattern
+      @pattern ||= options[:subdir] ? Regexp.new("^" + @options[:subdir]) : /.*/
+    end
+
+    def subdir
+      @options[:subdir] || ""
+    end
+
+    def range
+      "#{@base}...#{@to}"
+    end
+
+    def shortstat
+      `git diff --shortstat --summary #{range} #{subdir}`.split(/\n/)
```

Comment:
When running deploy-complexity inside of NoRedInk/monolith I see the pull requests are filtered correctly by subdir! However we get this error:
`fatal: ambiguous argument 'monolith/': unknown revision or path not in the working tree.`

We were able to `puts` and find that at least shortstat is giving off this error, if not others.

What we did
We changed the gemfile two different ways:
1. to point at your branch we changed the `tag` section for deploy-complexity to `branch: your-branch`
2. we also started playing around locally and changed the newly added `branch` section to `path: /direct/path/to/deploy-complexity`

With this we were able to `bundle` inside of NoRedInk/monolith and get our new version of the gem!

Running `bin/deploy-complexity.rb --subdir 'monolith/'` gave us the error above

Reply:
Weird, haven't seen that when running the script using `--git-dir` from the `deploy-complexity` repo. Thank you for the report, will take another look tomorrow!

Reply:
I could reproduce the error super easily, thank you for the clear instructions! I didn't account for the command to be ran from a subdirectory. Fixed now, would love it if you could take another look!

Reply:
Just verified this is now resolved.